cmd/nic-configuration-daemon/main.go

```diff
@@ -1,5 +1,54 @@
 package main
 
+import (
+	"flag"
+	"os"
+
+	"github.com/Mellanox/nic-configuration-operator/api/v1alpha1"
+	"github.com/Mellanox/nic-configuration-operator/pkg/daemon"
+	"github.com/Mellanox/nic-configuration-operator/pkg/host"
+	ncolog "github.com/Mellanox/nic-configuration-operator/pkg/log"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
 func main() {
+	ncolog.BindFlags(flag.CommandLine)
+	flag.Parse()
+	ncolog.InitLog()
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Log.Error(err, "failed to in-cluster configuration")
+		os.Exit(1)
+	}
+
+	err = v1alpha1.AddToScheme(scheme)
+	if err != nil {
+		log.Log.Error(err, "failed to load sriov network CRDs to scheme")
+		os.Exit(1)
+	}
+
+	kubeClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		log.Log.Error(err, "failed to create clientset")
+		os.Exit(1)
+	}
+
+	// Get the pod name and namespace from the environment variables
+	nodeName := os.Getenv("NODE_NAME")
+
+	hostUtils := host.NewHostUtils()
+	configDaemon := daemon.New(kubeClient, nodeName, host.NewHostManager(nodeName, hostUtils), hostUtils)
```

Comment:
daemon is going to have several parts ("controllers")
what do you think about using controller manager for it ?

you can register reconcilers as well as add "runnable" components if needed.

Reply:
Changed daemon to be a NicDevice reconciler, now it uses a controller manager